src/qt/mrcmodel.cpp

```diff
@@ -259,6 +261,18 @@ void MRCModel::refresh() EXCLUSIVE_LOCKS_REQUIRED(cs_main)
     // Store this locally so we don't have to get this from the client model, which takes another lock on cs_main.
     m_block_height = pindexBest->nHeight;
 
+    // Do a balance check before anything else.
+    if (m_wallet_model) {
+        CAmount mature_balance = m_wallet_model->getBalance();
+
+        if (mature_balance < COIN) {
+            m_mrc_error |= true;
+            m_mrc_status = MRCRequestStatus::INSUFFICIENT_MATURE_FUNDS;
+            m_mrc_error_desc = tr("You must have a mature balance of at least 1 GRC to submit an MRC.");
```

Comment:
should this be "submit a MRC" instead of "an"?

Reply:
Well... that is debatable. Depends on whether you say MRC or Manual Rewards Claim...

Reply:
I think it should remain as is. See https://www.roshreview.com/blog/when-to-use-a-or-an-before-acronyms/?utm_source=adwords&utm_medium=ppc&utm_campaign=DSA+-+Rosh+-+18-45%2Bunk+-+US&utm_term=&hsa_cam=16022828280&hsa_src=g&hsa_mt=&hsa_net=adwords&hsa_kw=&hsa_tgt=dsa-43245954176&hsa_grp=127907509850&hsa_ver=3&hsa_ad=578568922778&hsa_acc=1412339070&gclid=CjwKCAiA5Y6eBhAbEiwA_2ZWIbgkSp74YFP-bnBFT3gbXBjwjp9e62OE-leArL4qaONfrXKIUuM7RRoCbz4QAvD_BwE